osquery/config/config.cpp

```diff
@@ -525,6 +528,120 @@ void stripConfigComments(std::string& json) {
   json = sink;
 }
 
+void translateYamlNodeToJson(
+    const YAML::Node& node,
+    rapidjson::Writer<rapidjson::StringBuffer>& convertedJsonWriter) {
+  switch (node.Type()) {
+  case YAML::NodeType::Sequence:
+    convertedJsonWriter.StartArray();
+    for (size_t i = 0; i < node.size(); i++) {
+      translateYamlNodeToJson(node[i], convertedJsonWriter);
+    }
+    convertedJsonWriter.EndArray();
+    break;
+
+  case YAML::NodeType::Map:
+    convertedJsonWriter.StartObject();
+    for (YAML::const_iterator it = node.begin(); it != node.end(); ++it) {
+      convertedJsonWriter.Key(it->first.as<std::string>().c_str());
+      translateYamlNodeToJson(it->second, convertedJsonWriter);
+    }
+    convertedJsonWriter.EndObject();
+    break;
+
+  case YAML::NodeType::Null:
+  case YAML::NodeType::Undefined:
+    convertedJsonWriter.Null();
+    break;
+
+  case YAML::NodeType::Scalar:
+    auto nodeVal = node.as<std::string>();
+
+    if (nodeVal == "~" || nodeVal == "null" || nodeVal == "Null" ||
+        nodeVal == "NULL") {
+      convertedJsonWriter.Null();
+    } else if (nodeVal == "true" || nodeVal == "True" || nodeVal == "TRUE") {
+      convertedJsonWriter.Bool(true);
+    } else if (nodeVal == "false" || nodeVal == "False" || nodeVal == "FALSE") {
+      convertedJsonWriter.Bool(false);
+    } else {
+      if (node.Tag() != "!") {
+        const char* end = nodeVal.c_str() + nodeVal.length();
+        char* pos = nullptr;
+
+        // hex strings
+        if (boost::starts_with(nodeVal, "0x")) {
+          auto i = strtoll(nodeVal.c_str() + 2, &pos, 16);
+          if (pos == end) {
+            convertedJsonWriter.Int64(i);
+            break;
+          }
+        }
+
+        // octal strings
+        if (boost::starts_with(nodeVal, "0o")) {
+          auto i = strtoll(nodeVal.c_str() + 2, &pos, 8);
+          if (pos == end) {
+            convertedJsonWriter.Int64(i);
+            break;
+          }
+        }
+
+        // try parsing it as a long
+        auto i = strtoll(nodeVal.c_str(), &pos, 10);
```

Comment:
Is it useful to support decimal annotated integers? ([YAML Spec - section 2.4](http://yaml.org/spec/1.2/spec.html#id2761292))

For example: +1234. I'm not sure if strtoll supports that or not.